src/MassTransit/DependencyInjection/DependencyInjection/GenericRequestClient.cs

```diff
@@ -123,12 +123,11 @@ public Task<Response<T>> GetResponse<T>(object values, CancellationToken cancell
 
         static IRequestClient<TRequest> GetRequestClient(IServiceProvider provider)
         {
-            var consumeContext = provider.GetRequiredService<ScopedConsumeContextProvider>().GetContext();
-
             var clientFactory = provider.GetService<IScopedClientFactory>();
             if (clientFactory != null)
                 return clientFactory.CreateRequestClient<TRequest>();
 
+            var consumeContext = provider.GetRequiredService<Bind<IMediator, IScopedConsumeContextProvider>>().Value.GetContext();
```

Comment:
Only call this after `mediator != null` check to avoid expensive lookup. It could actually fail since it's required and might not even be registered.